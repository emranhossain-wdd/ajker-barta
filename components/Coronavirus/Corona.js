import { useRouter } from 'next/router';
import React from 'react';

const Corona = ({news}) => {

const router = useRouter()   
    const {heading,description,images,publishedDate}=news
    return (
        <div onClick={() => router.push(`/news/${news._id}`)} className="rounded overflow-hidden shadow-lg cursor-pointer">
        <img className="w-full" src={images.img1} alt="Mountain"/>
        <div className="px-6 py-4">
          <div className="font-bold text-xl mb-2">{heading}</div>
          <p className="text-gray-700 text-base">
          {description[0].slice(0,100)}
          </p>
        </div>
        <div className="px-6 pt-4 pb-2">
          <span className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2 mb-2">{publishedDate}</span>
        </div>
      </div>
    );
};

export default Corona;