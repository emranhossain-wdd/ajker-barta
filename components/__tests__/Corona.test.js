import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Corona from '../Coronavirus/Corona'

it('render SportAndVoting component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Corona />, div)
})
it('maching snapshot', () => {
  const tree = renderer.create(<Corona />)
  expect(tree).toMatchSnapshot()
})