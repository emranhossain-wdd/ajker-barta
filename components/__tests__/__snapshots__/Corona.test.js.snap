// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="rounded imageEffect overflow-hidden shadow-md cursor-pointer"
  onClick={[Function]}
>
  <div
    className="w-full overflow-hidden"
  >
    <img
      alt="Mountain"
      className="w-full"
    />
  </div>
  <div
    className="px-3 py-3"
  >
    <div
      className="font-bold text-lg mb-1 leading-6 hover:text-red-600 transition-colors duration-300"
    />
    <p
      className="text-gray-700 text-sm"
    />
    <div
      className="mt-4"
    />
  </div>
</div>
`;
